From 0a2ff9391d81586e0faf783fe2a34218a04beadb Mon Sep 17 00:00:00 2001
From: Potin Lai <potin.lai@quantatw.com>
Date: Tue, 10 Jun 2025 09:11:21 +0800
Subject: [PATCH 5/6] configurations: cx7_ocp: Switch to external sensor for
 temperature reading

Catalina hardware design does not support using `pldmd` to handle PLDM
transactions (i2c mux with BE NICs). To work around this limitation,
this change switches to using an external sensor interface. A background
service is introduced that periodically invokes `pldmtool` to query the
NIC temperature and updates the external sensor accordingly.

Signed-off-by: Potin Lai <potin.lai@quantatw.com>
Change-Id: I8a6fcb383b4d020b10e48543314a9f1b865ec0de
---
 configurations/ocp/cx7_ocp.json | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/configurations/ocp/cx7_ocp.json b/configurations/ocp/cx7_ocp.json
index 5ffa735..e9c46f9 100644
--- a/configurations/ocp/cx7_ocp.json
+++ b/configurations/ocp/cx7_ocp.json
@@ -7,13 +7,11 @@
             "Type": "EEPROM"
         },
         {
-            "Address": "0x1f",
-            "Bus": "$bus",
-            "Labels": [
-                "temp2"
-            ],
             "Name": "NIC$bus % 2 TEMP_C",
-            "Name1": "NIC$bus % 2 TEMP_C",
+            "Units": "xyz.openbmc_project.Sensor.Value.Unit.DegreesC",
+            "MinValue": -255,
+            "MaxValue": 255,
+            "Timeout": 30.0,
             "Thresholds": [
                 {
                     "Direction": "less than",
@@ -32,7 +30,7 @@
                     "Value": 105
                 }
             ],
-            "Type": "TMP421"
+            "Type": "ExternalSensor"
         }
     ],
     "Name": "OCP CX7 NIC $bus % 2",
-- 
2.47.3

